api/maven-api-core/src/main/java/org/apache/maven/api/services/ProblemCollector.java

```diff
@@ -212,20 +229,22 @@ class Impl<P extends BuilderProblem> implements ProblemCollector<P> {
         private final AtomicInteger totalCount;
         private final ConcurrentMap<BuilderProblem.Severity, LongAdder> counters;
         private final ConcurrentMap<BuilderProblem.Severity, List<P>> problems;
+        private final Predicate<? super P> filter;
 
         private static final List<BuilderProblem.Severity> REVERSED_ORDER = Arrays.stream(
                         BuilderProblem.Severity.values())
                 .sorted(Comparator.reverseOrder())
                 .toList();
 
-        private Impl(int maxCountLimit) {
+        private Impl(int maxCountLimit, final Predicate<? super P> filter) {
```

Comment:
```suggestion
       private Impl(int maxCountLimit, Predicate<? super P> filter) {
```
no need for final in parameters